cmd/buildah/unshare.go

```diff
@@ -57,7 +57,7 @@ func bailOnError(err error, format string, a ...interface{}) {
 
 func maybeReexecUsingUserNamespace(cmdName string, evenForRoot bool) {
 	// If we've already been through this once, no need to try again.
-	if unshare.IsRootless() {
+	if os.Geteuid() == 0 && unshare.IsRootless() {
```

Comment:
should this be '!= 0'?  If so,isn't that covered in unshare.IsRootless already?

 		isRootless = os.Geteuid() != 0 || os.Getenv(UsernsEnvName) != ""

Reply:
no, it should be == 0, we want to skip the test here only if we are already running as root user.

Inside unshare.IsRootless then we might return early if the env variable is set

Reply:
OK, the title for the PR threw me off, just a a dyslexic moment on my end. I see what's happening now.